zomes/wiki/code/src/page.rs

```diff
@@ -95,75 +84,95 @@ pub fn page_def() -> ValidatingEntryType {
 
     )
 }
-pub fn create_page_if_non_existent(title: String) -> ZomeApiResult<Address> {
-    let address = Page::initial(title.clone()).entry().address();
-    match hdk::get_entry(&address)? {
-        None => {
-            let page_anchor = utils::anchor("wiki_pages", "all_pages")?;
-            hdk::utils::commit_and_link(
-                &Page::initial(title.clone()).entry(),
-                &page_anchor,
-                "anchor->Page",
-                &title,
-            )
-        }
-        Some(_) => Ok(address),
+pub fn create_page_if_non_existent(
+    title: String,
+    sections: Vec<Address>,
+    timestamp: String,
+) -> ZomeApiResult<Address> {
+    let anchor_address = holochain_anchors::anchor("wiki_pages".to_string(), title.clone())?;
+    if let Some(address) = hdk::get_links(
+        &anchor_address,
+        LinkMatch::Exactly("anchor->Page"),
+        LinkMatch::Any,
+    )?
+    .addresses()
+    .pop()
+    {
+        Ok(address)
+    } else {
+        let page_entry = Page::from(title.clone(), sections, timestamp).entry();
+
+        hdk::utils::commit_and_link(&page_entry, &anchor_address, "anchor->Page", &title)
     }
 }
-
 pub fn create_page_with_sections(
-    sections: Vec<Section2>,
+    sections: Vec<Section>,
     title: String,
-    timestamp: u64,
+    timestamp: String,
 ) -> ZomeApiResult<String> {
-    let page_address = create_page_if_non_existent(title.clone())?;
-    let sections: Vec<Address> = sections
-        .into_iter()
-        .map(|element| {
-            hdk::api::commit_entry(&Section::from(element, page_address.clone()).entry())
-        })
-        .filter_map(Result::ok)
-        .collect();
+    let anchor_address = holochain_anchors::anchor("wiki_pages".to_string(), title.clone())?;
 
-    hdk::api::update_entry(
-        Page::from(title.clone(), sections, Some(timestamp)).entry(),
-        &page_address,
+    create_page_if_non_existent(
+        title.clone(),
+        sections
+            .into_iter()
+            .map(|section| hdk::api::commit_entry(&section.from(anchor_address.clone()).entry()))
```

Comment:
Change all instances of `hdk::api::*` to `hdk::*`

Reply:
For readibility: Also split this code block into two: first create all sections and get the list of addresses, and then create the page